molvis/molvisgl/src/processors/molecularrenderer.cpp

```diff
@@ -207,13 +258,15 @@ void MolecularRenderer::process() {
         }
     }();
 
-    if (meshes_.empty() || inport_.isChanged() || updateColorMap) {
+    if (meshes_.empty() || inport_.isChanged() || updateColorMap || brushing_.isChanged()) {
         meshes_.clear();
-        size_t pickingId = atomPicking_.getPickingId(0);
+        const size_t pickingId = atomPicking_.getPickingId(0);
+        size_t offset = 0;
         for (auto structure : inport_) {
-            meshes_.push_back(createMesh(
-                *structure, {coloring_, atomColormap_, aminoColormap_, fixedColor_}, pickingId));
-            pickingId += structure->atoms().positions.size();
+            meshes_.push_back(createMesh(*structure,
```

Comment:
here we could definitely reuse the mesh since it is internal to the renderer...

Reply:
This is already the case, hence the use of `mesh_` and updating it only when necessary.